src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresDatatypeIT.java

```diff
@@ -85,10 +91,96 @@ public void after() throws SQLException {
     log.info("Dropped table");
   }
 
+  /**
+   * Verifies that even when the connector encounters exceptions that would cause a connection
+   * with an invalid transaction, the connector sends only the errant record to the error
+   * reporter and establishes a valid transaction for subsequent correct records to be sent to
+   * the actual database.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testPrimaryKeyConstraintsSendsToErrorReporter() throws Exception {
+    props.put(ERRORS_TOLERANCE_CONFIG, ToleranceType.ALL.value());
+    props.put(DLQ_TOPIC_NAME_CONFIG, DLQ_TOPIC_NAME);
+    props.put(DLQ_TOPIC_REPLICATION_FACTOR_CONFIG, "1");
+    props.put(MAX_RETRIES, "2");
+
+    createTableWithPrimaryKey();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
+    final Schema schema = SchemaBuilder.struct().name("com.example.Person")
+        .field("firstname", Schema.STRING_SCHEMA)
+        .field("lastname", Schema.STRING_SCHEMA)
+        .build();
+    final Struct struct = new Struct(schema)
+        .put("firstname", "Christina")
+        .put("lastname", "Brams");
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct));
+    connect.kafka().produce(tableName, null, kafkaValue);
+    // Send the same record for a PK collision
+    connect.kafka().produce(tableName, null, kafkaValue);
+
+    // Now, create and send another normal record
+    Struct struct2 = new Struct(schema)
+        .put("firstname", "Brams")
+        .put("lastname", "Christina");
+
+    kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct2));
+    connect.kafka().produce(tableName, null, kafkaValue);
+
+    try {
+      // Consume the number of records produced since this is the upper bound of records that
+      // could be sent to the error reporter.
+      ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(3,
+          CONSUME_MAX_DURATION_MS, DLQ_TOPIC_NAME);
```

Comment:
Doesn't this force the test to run for `CONSUME_MAX_DURATION_MS`? It'd be nice to avoid bloat in our integration test runtimes if possible.

There are a few alternatives we might consider here:
- Wait for the connector to commit offsets for every record we've written to it, then do our consumer poll (which would then return up to 500 records, the default for the [max.poll.records](https://kafka.apache.org/27/documentation.html#consumerconfigs_max.poll.records) property, in a single call)
- Wait for the connector to write the second record value to the table first, then poll the consumer (although if for some reason there's a large pause between when the framework is able to write a first and second record to the DLQ, this may miss that second record; however, the current approach also misses this case, and given that it's not super likely, I think that's acceptable)
- Consume the expected number of records (in this case, I believe that's 1?) initially with the `CONSUME_MAX_DURATION_MS` timeout, then do a follow-up consume with a much lower timeout (like 5 seconds tops)

Reply:
Also, nit: right now the `records` variable is unused and this section can just be changed to:
```suggestion
      connect.kafka().consume(3, CONSUME_MAX_DURATION_MS, DLQ_TOPIC_NAME);
```

(assuming any of the suggestions above don't change things)